Lib/test/test_re.py

```diff
@@ -918,44 +918,46 @@ def assertMatch(self, pattern, text, match=None, span=None,
         self.assertEqual(m.span(), span)
 
     def test_re_escape(self):
-        alnum_chars = string.ascii_letters + string.digits + '_'
+        literal_chars = string.ascii_letters + string.digits + '!"%&\',/:;<=>@_`~'
         p = ''.join(chr(i) for i in range(256))
         for c in p:
-            if c in alnum_chars:
-                self.assertEqual(re.escape(c), c)
-            elif c == '\x00':
-                self.assertEqual(re.escape(c), '\\000')
-            else:
-                self.assertEqual(re.escape(c), '\\' + c)
             self.assertMatch(re.escape(c), c)
+            self.assertMatch('[' + re.escape(c) + ']', c)
+            self.assertMatch('(?x)' + re.escape(c), c)
         self.assertMatch(re.escape(p), p)
+        for c in '-.]{}':
+            self.assertEqual(re.escape(c)[:1], '\\')
+        literal_chars = (string.ascii_letters + string.digits +
```

Comment:
This redefines `literal_chars` to the same as it was before. Either delete, or split the test, or hoist to a suite level constant?